rpmlb/recipe.py

```diff
@@ -23,22 +25,39 @@ def __init__(self, file_path, recipe_id):
         self.num_of_package = len(self.recipe['packages'])
 
     def each_normalized_package(self):
+        """Present recipe packages in normalized form.
+
+        Yields:
+            Dictionary with the package's data.
+
+        Common keys in the yielded dictionary:
+            - name: The package's name
+            - booststrap: Numerical position in the bootstrapping
+                sequence, or None if no bootstrap is necessary.
+            - macros, replaced_macros: Mapping of macro name to (new)
+                macro body.
+        """
+
         packages = self.recipe['packages']
+        bootstrap_map = self._count_bootstrap_sequences()
+
         for package in packages:
-            package_dict = {}
-            # String or hash
-            if isinstance(package, str):
-                package_dict['name'] = package
-            elif isinstance(package, dict):
-                keys = list(package.keys())
-                name = keys[0]
-                if not package[name]:
-                    raise ValueError('Recipe package %s is invalid format.' %
-                                     name)
-                package_dict = package[name]
-                package_dict['name'] = name
-            else:
-                raise ValueError('package is invalid.')
+            name = self._package_name(package)
+            bootstrap_position = next(bootstrap_map[name], None)
+
+            package_dict = {
+                'name': name,
+                'bootstrap': bootstrap_position,
+            }
+
+            if isinstance(package, Mapping):
+                # package should be one-item dictionary
+                if len(package) != 1:
+                    msg = 'Invalid package data: {}'.format(package)
```

Comment:
I prefer "message" than "msg", Because as a variable to describe Message, "message" is used in this project.

```
$ grep -r msg -r rpmlb
$ grep -r message -r rpmlb
rpmlb/builder/base.py:            message = (
rpmlb/builder/base.py:            LOG.info(message)
rpmlb/builder/base.py:                message = 'pacakge_dict: {0}, num: {1}, work_dir: {2}'.format(
rpmlb/builder/base.py:                error = RuntimeError(message)
rpmlb/utils.py:        # Use absolute path to show it on FileNotFoundError message.
rpmlb/__init__.py:LOG_FORMAT = '%(asctime)s %(levelname)-8s %(name)-26s %(message)s'
rpmlb/downloader/base.py:            message = (
rpmlb/downloader/base.py:            LOG.info(message)
```